src/updater/formatPlugin.ts

```diff
@@ -0,0 +1,40 @@
+import type { ProjectManifest } from '@pnpm/types'
+
+export function createFormat<Content>(format: FormatPlugin<Content>) {
+  return format
+}
+
+export interface BaseFormatPlugins {
+  [extension: string]: FormatPlugin<unknown>
+}
+
+export interface FormatPlugin<Content> {
+  /** Called only if file exists */
+  read(options: FormatPluginFnOptions): PromiseOrValue<Content>
+
+  /** `actual` is `null` when file doesn't exist */
+  update(
+    actual: Content | null,
+    updater: Updater<Content>,
+    options: FormatPluginFnOptions
+  ): PromiseOrValue<Content | null>
+
+  /** Called only if check for equality is required (`actual != null` & `expected != null`) */
+  equal(expected: Content, actual: Content, options: FormatPluginFnOptions): PromiseOrValue<boolean>
+
+  /** Called only if write is required (`--test` isn't specified, `expected != null` and `expected` is not equal to `actual`) */
+  write(expected: Content, options: FormatPluginFnOptions): PromiseOrValue<void>
+}
+
+export interface FormatPluginFnOptions {
+  // checkOnly: boolean
```

Comment:
we can pass some boolean variable here kinda `checkOnly` or `shouldUpdate` that reflects `opts?.test`